### ISSUE TITLE
Missing XML Quoting support in vcxproj generator

### ISSUE THREAD

**Issue by miv-ableton** [fastbuild/fastbuild#1028]
vcxproj files generated by FastBuild may be invalid if the content includes `<` or `>`. In my company's proprietary codebase, one project uses these characters in the tag `NMakePreprocessorDefinitions`, and this makes the generated vcxproj unavailable to VS.

Looking at Fasbuild's code, it seems that detecting `<` and `>` and replacing them with `&lt;` and `&gt;` should be done in `Code/Tools/FBuild/FBuildCore/Helpers/VSProjectGenerator.cpp`, in the method:

```cpp
void VSProjectGenerator::WritePGItem( const char * xmlTag, const AString & value )
```

I'm sorry I can't submit a PR. I'm unable to figure out how to make the Fastbuild source code build.

**Comment by ffulin** [comment]
Thanks for reporting this problem. Your proposal seems reasonable.

Seems like we might want to escape quotes and ampersand as well.

There is an EscapeArgument function in XCodeProjectGenerator that can be moved to the shared based class and leveraged in the VCXProj generation.

**Comment by ffulin** [comment]
I've implemented escaping for VCXProj options globally here: https://github.com/fastbuild/fastbuild/commit/9dba1a3e446c7ae0e6ef08dd8bfd850003f6c39d

In the end, I didn't combine/unify with the EscapeArgument function as that behaves on xml tags for XCode projects as opposed to xml data between tags. After some experimentation, escaping ```<```, ```>``` and ```&``` was needed to behave consistently with edits performed via the Visual Studio UI.

This fix will be in the next release (v1.13). I'll keep this issue open until that version is released.

**Comment by ffulin** [comment]
v1.13 has now been released and includes this fix. Thanks again for reporting the bug.
